### ISSUE TITLE
spec file for bbswitch not compatible with kernel 5.4.x

### ISSUE THREAD

**Issue by slartibart70** [chenxiaolong/fedora-packages#9]
When using a kernel 5.4.x and trying to compile (indirectly by akmod) bbswitch, we get no kernel module.
Reasons:
SUBDIRS parameter is not supported any more,
i think a simple fix in the bbswitch-kmod.spec file would be enough?

with 5.3.16 you already get a warning in the build log for bbswitch:

2020/01/03 17:57:10 akmodsbuild: + make -j8 -C /usr/src/kernels/5.3.16-300.fc31.x86_64 SUBDIRS=/tmp/akmodsbuild.0kiWMiOq/BUILD/bbswitch-kmod-0.8/_kmod_build_5.3.16-300.fc31.x86_64 modules
2020/01/03 17:57:10 akmodsbuild: make: Entering directory '/usr/src/kernels/5.3.16-300.fc31.x86_64'
2020/01/03 17:57:10 akmodsbuild: Makefile:213: ================= WARNING ================
2020/01/03 17:57:10 akmodsbuild: Makefile:214: 'SUBDIRS' will be removed after Linux 5.3
2020/01/03 17:57:10 akmodsbuild: Makefile:215:
2020/01/03 17:57:10 akmodsbuild: Makefile:216: If you are building an individual subdirectory
2020/01/03 17:57:10 akmodsbuild: Makefile:217: in the kernel tree, you can do like this:
2020/01/03 17:57:10 akmodsbuild: Makefile:218: $ make path/to/dir/you/want/to/build/
2020/01/03 17:57:10 akmodsbuild: Makefile:219: (Do not forget the trailing slash)
2020/01/03 17:57:10 akmodsbuild: Makefile:220:
2020/01/03 17:57:10 akmodsbuild: Makefile:221: If you are building an external module,
2020/01/03 17:57:10 akmodsbuild: Makefile:222: Please use 'M=' or 'KBUILD_EXTMOD' instead
2020/01/03 17:57:10 akmodsbuild: Makefile:223: ==========================================

with kernel 5.4.x you get (strange) errors like so:

2020/01/03 17:34:34 akmodsbuild: + make -j8 -C /usr/src/kernels/5.4.5-300.fc31.x86_64 SUBDIRS=/tmp/akmodsbuild.F7BntqoM/BUILD/bbswitch-kmod-0.8/_kmod_build_5.4.5-300.fc31.x86_64 modules
2020/01/03 17:34:34 akmodsbuild: make: Entering directory '/usr/src/kernels/5.4.5-300.fc31.x86_64'
2020/01/03 17:34:34 akmodsbuild: HOSTCC  scripts/basic/fixdep
2020/01/03 17:34:34 akmodsbuild: scripts/basic/fixdep.c:410:1: fatal error: opening dependency file scripts/basic/.fixdep.d: Permission denied
2020/01/03 17:34:34 akmodsbuild: 410 | }
2020/01/03 17:34:34 akmodsbuild: | ^
2020/01/03 17:34:34 akmodsbuild: compilation terminated

**Comment by slartibart70** [comment]
just checked,
a change for the make command from SUBDIRS= to M= is sufficient.

**Comment by cowboysmall** [comment]
Seeing the same issue.

**Comment by cowboysmall** [comment]
Any idea when this might be resolved? 

@slartibart70 - if possible could you describe how to fix this locally? I'm not sure how to proceed as I can't find the relevant source code / make file.

edit: should I clone the repository and make whatever changes are necessary? Can I build the rpm locally and install it locally?

**Comment by slartibart70** [comment]
@cowboysmall 
I did a workaround to fix it on my system... like so (extract /usr/src/akmods/bbswitch-kmod-0.8-2.fc31.src.rpm, modify spec file and recreate srpm)
```
rpmdev-setuptree
rpm2cpio /usr/src/akmods/bbswitch-kmod-0.8-2.fc31.src.rpm | cpio -idmv
cp v0.8.tar.gz /home/user/rpmbuild/SOURCES/
cp bbswitch-kmod.spec /home/user/rpmbuild/SPECS/
modify /home/user/rpmbuild/SPECS/bbswitch-kmod.spec; change in Build-section: make SUBDIRS= to M=
rpmbuild -ba bbswitch-kmod.spec
put the newly created /home/user/rpmbuild/SRPMS/bbswitch-kmod-0.8-2.fc31.src.rpm back to /usr/src/akmods location
chcon --reference nvidia-kmod-440.44-1.fc31.src.rpm bbswitch-kmod-0.8-2.fc31.src.rpm (or set selinux context by hand to 'system_u:object_r:usr_t:s0')
stop bumblebeed service, 
run akmods-ostree-post 
run /usr/sbin/akmods --from-akmod-posttrans --akmod bbswitch
check if module loads with 'bumblebeed -vv', if so, Ctrl-C and restart bumblebeed.service
```

**Comment by chenxiaolong** [comment]
Fixed in #10 and pushed to copr as `bbswitch-kmod-0.8-3.fc##`. Thanks!

**Comment by cowboysmall** [comment]
@slartibart70 - thanks for that!



> @cowboysmall
> I did a workaround to fix it on my system... like so (extract /usr/src/akmods/bbswitch-kmod-0.8-2.fc31.src.rpm, modify spec file and recreate srpm)
>
